core/src/test/scala/unit/kafka/server/LogDirFailureTest.scala

```diff
@@ -181,8 +181,21 @@ class LogDirFailureTest extends IntegrationTestHarness {
     // Wait for producer to update metadata for the partition
     TestUtils.waitUntilTrue(() => {
       // ProduceResponse may contain KafkaStorageException and trigger metadata update
-      producer.send(record)
-      producer.partitionsFor(topic).asScala.find(_.partition() == 0).get.leader().id() != leaderServerId
+      {
```

Comment:
Thanks for the update!

It seems that the approach before this PR will also retry producer.send(..) until metadata is refreshed. We can just add `producer.send(record).get(6000, TimeUnit.MILLISECONDS)` after `TestUtils.waitUntilTrue(//send message)` (line 182) and before consuming messages. This approach seems a bit simpler since it adds only 1 line. Do you think that is better?

Otherwise LGTM if you like the existing approach in this PR. If so, can you address comments from @dhruvilshah3?

Reply:
Ah I didn't understand what you meant at first but that makes a lot of sense. I wen't with that change instead. Thanks!